src/utils.cpp

```diff
@@ -1481,4 +1481,30 @@ std::vector<int> get_all_running_pids()
   return pids;
 }
 
+// BPF verifier rejects programs with names containing certain characters, use
+// this function to replace every character not valid for C identifiers by '_'
+std::string sanitiseBPFProgramName(std::string const &name)
+{
+  std::string sanitised_name = name;
+  std::replace_if(
+      sanitised_name.begin(),
+      sanitised_name.end(),
+      [](char c) { return !isalnum(c) && c != '_'; },
+      '_');
+
+  // Kernel KSYM_NAME_LEN is 128 until 6.1
+  // If we'll exceed the limit, hash the string and cap at 128.
+  if (sanitised_name.size() > 128)
+  {
+    size_t hash = std::hash<std::string>{}(sanitised_name);
+
+    // std::hash returns size_t, so we reserve 2*sizeof(size_t)+1 characters
```

Comment:
Hmm, still unclear why `2*` is necessary

Reply:
Because each byte is encoded with 2 chars. So, `size_t` is normally 8 bytes which means we need 16 chars to represent is as a hex string.

Reply:
ack